pandas/core/arrays/categorical.py

```diff
@@ -1980,9 +1980,9 @@ def _reverse_indexer(self) -> Dict[Hashable, np.ndarray]:
         """
         categories = self.categories
         r, counts = libalgos.groupsort_indexer(
-            self.codes.astype("int64", copy=False), categories.size
+            libalgos.ensure_platform_int(self.codes), categories.size
```

Comment:
IIUC ensure_platform_int should be imported from pandas.core.dtypes.common see #40567

Reply:
sure, mind if i do this in the next pass?  have a branch on deck

Reply:
ok to do later, but maybe should rebase this PR anyhow as @simonjayhawkins change was merged

Reply:
sure